skunkworks_crow/src/main/java/org/odk/share/utilities/PermissionUtils.java

```diff
@@ -56,6 +59,30 @@ public static void showLocationAlertDialog(Activity targetActivity) {
         builder.setCancelable(false);
         builder.show();
     }
+
+    /**
+     * Showing an alert dialog and send users to app info page.
+     */
+    public static void showAppInfo(Activity targetActivity, String packageName) {
+        AlertDialog.Builder builder = new AlertDialog.Builder(targetActivity);
+        builder.setMessage(R.string.permission_open_info);
+
+        builder.setPositiveButton(targetActivity.getString(R.string.permission_open_info_button), (DialogInterface dialog, int which) -> {
+            Intent intent = new Intent();
+            intent.setAction(Settings.ACTION_APPLICATION_DETAILS_SETTINGS);
+            Uri uri = Uri.fromParts(SCHEME, packageName, null);
+            intent.setData(uri);
+            targetActivity.startActivity(intent);
```

Comment:
There is an issue over here when the user moves to app settings, and no matter what he does there and after doing all the activity on app settings when he presses back he goes back to the `Send data` screen with empty view.
You should check whenever an activity is resumed, whether user actually left application to actually play with the app settings, if the answer is yes then check if we've enough permission to do the transmission, if not then again display the same dialog.

Reply:
Fixed, I tested in android 9 and non-android 9 devices, no issues, but I think it still needs more tests.